server/rest/src/main/java/org/hawkular/apm/server/rest/AnalyticsHandler.java

```diff
@@ -454,6 +454,42 @@ public void getTraceCompletionCount(
 
     }
 
+    @POST
+    @Path("trace/completion/count")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(
+            value = "Get the trace completion count",
+            response = Long.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 500, message = "Internal server error") })
+    public void getTraceCompletionCount(
+            @Context SecurityContext context, @HeaderParam("Hawkular-Tenant") String tenantId,
+            @Suspended final AsyncResponse response,
+            @ApiParam(required = true,
+                    value = "query criteria") Criteria criteria) {
+
+        try {
+            log.tracef("Get trace completion count for criteria [%s]", criteria);
+
+            long count = analyticsService.getTraceCompletionCount(
+                    securityProvider.validate(tenantId, context.getUserPrincipal().getName()), criteria);
+
+            log.tracef("Got trace completion count: criteria [%s] = [%s]", criteria, count);
+
+            response.resume(Response.status(Response.Status.OK).entity(count).type(APPLICATION_JSON_TYPE)
+                    .build());
+
+        } catch (Throwable e) {
```

Comment:
I think that here is better to catch checked exceptions and for unchecked use exception mappers.